rust/src/tests/rmdir.rs

```diff
@@ -29,3 +29,44 @@ fn changed_time_parent_success(ctx: &mut TestContext) {
         });
     });
 }
+
+/// Dummy mountpoint to check that rmdir returns EBUSY when using it on a mountpoint.
+struct DummyMnt {
+    path: PathBuf,
+}
+
+impl DummyMnt {
+    fn new(ctx: &mut TestContext) -> anyhow::Result<Self> {
+        // We don't really care about a specific type of file system here, the directory just have to be a mount point
+        let from = ctx.create(crate::runner::context::FileType::Dir)?;
+        let path = ctx.create(crate::runner::context::FileType::Dir)?;
+        let mut mount = Command::new("mount");
+
+        if cfg!(target_os = "linux") {
+            mount.arg("--bind");
+        } else {
+            mount.args(["-t", "nullfs"]);
+        }
+
+        let result = mount.arg(&from).arg(&path).output()?;
+        debug_assert!(result.status.success());
+
+        Ok(Self { path })
+    }
+}
+
+impl Drop for DummyMnt {
+    fn drop(&mut self) {
+        let umount = Command::new("umount").arg(&self.path).output();
+        debug_assert!(matches!(umount, Ok(res) if res.status.success()));
```

Comment:
It's a bad idea to panic during Drop, because the thread might already be panicing.  Double-panicing screws up the stack trace and prevents other Drop handlers from running.  I suggest this instead.
```suggestion
        if !std::thread::panicking() {
            assert!(matches!(umount, Ok(res) if res.status.success()));
        }
```

Reply:
Oh yes, you're right, I forgot that it causes a panic!